src/components/dropdown/dropdown.tsx

```diff
@@ -63,18 +63,28 @@ export class Dropdown implements InteractiveComponent, OpenCloseComponent, Float
    */
   @Prop({ reflect: true, mutable: true }) active = false;
 
+  @Watch("active")
+  activeHandler(value: boolean): void {
+    if (!this.disabled) {
```

Comment:
I think you can get rid of all the logic here and just have the handler set the open value. Let the open watch handler do the logic.